domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -497,6 +503,27 @@ private void init() throws Exception {
     }
 
     client = createPgPool(vertx, postgreSQLClientConfig);
+
+
+    vertx.setPeriodic(CHECK_FOR_QUERY_TIMEOUT_INTERVAL, PostgresClient::checkStaleConnections);
+
+  }
+
+  private static void checkStaleConnections(Long timerId) {
+    for (SQLConnection conn : activeConnections) {
+      if (conn.timeoutTime >= System.currentTimeMillis()) {
```

Comment:
There is no need to call System.currentTimeMillis() in each for iteration, you may use

    long now = System.currentTimeMillis()

before the for loop.